.github/workflows/build-and-test.yml

```
@@ -0,0 +1,27 @@
+name: Build and Test
+
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: "*"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16"
+      - uses: pnpm/action-setup@v2.0.1
+        with:
+          version: 8.4.0
+      - run: git checkout main && git checkout $GITHUB_SHA
+      - run: pnpm install --frozen-lockfile
+      - run: pnpm run build
+      - run: pnpm run lint
```

Please add a check-format script to the package.json to verify prettier format (`prettier --check .`). I generally don't use eslint for formatting, so it's a separate command:

```suggestion
      - run: pnpm run lint
      - run: pnpm run check-format
```